setup.cfg

```diff
@@ -1,19 +1,32 @@
 [flake8]
 ignore =
     # Conflicts with black
-    E203
+    E203 
     # Line break occurred before a binary operator. Update by W504 Line
     W503
     # Allow "import torch.nn.functional as F"
     N812
+    # Line length, handled separately by max-line-length
+    E501
+    # too many leading '#' for block comments
+    E266
+    
 
 per-file-ignores =
     # Unused imports in __init__.py are OK
     **/__init__.py:F401
     
 exclude = 
     *_pb2*,
+    .git,
+    __pycache__,
+    build,
+    dist,
+    .venv
 
-max-line-length = 100
+max-line-length = 120
 
 copyright-check = True
+
+# Enable specific checks or plugins
+select = B,C,E,F,W,T4,B9
```

Comment:
Didn't understand, what does this line do?

Reply:
select specifies specific error and warning codes or plugin checks to enable:

B: Bandit security checks (e.g., detecting insecure function use).
C: Cyclomatic complexity, used to flag overly complex functions.
E: PEP8 errors (e.g., style issues).
F: Pyflakes errors, like unused imports or undefined names.
W: PEP8 warnings (e.g., stylistic issues).
T4: Type checking from third-party tools (like mypy).
B9: Bugbear, for additional warnings about potentially error-prone code.

Reply:
It would be useful to have this comment block in `setup.cfg`.

Reply:
added comments in setup.cfg